crates/astria-sequencer/src/transaction/mod.rs

```diff
@@ -241,10 +272,20 @@ impl ActionHandler for UnsignedTransaction {
         state: &S,
         from: Address,
     ) -> anyhow::Result<()> {
+        // Transactions must match the chain id of the node.
+        let chain_id = state.get_chain_id().await?;
+        ensure!(
+            self.params.chain_id == chain_id.as_str(),
+            InvalidChainId(self.params.chain_id.clone())
+        );
+
```

Comment:
you can remove this - the mempool check is sufficient, no txs with invalid chain ids will make it here. the nonce check below is needed for strict equality before execution

Reply:
I think we need to protect against a validator running their node with a no op mempool. IE someone can modify code for their validator grab alternate transactions and propose?